### ISSUE TITLE
Refactor digest() similar to vdigest

### ISSUE THREAD

**Issue by eddelbuettel** [eddelbuettel/digest#115]
@mdequeljoe : Now that we have a new digest version [which is on its way to CRAN now] thanks to your vectorised approach with the function getter, I want to get back to what you first proposed in your PR.  I sort-of misunderstand the value of it, but it was also lost a little in that PR. It really did two things, and we re-focused on adding vectorised only.

Now, making `digest()` simpler given that you correctly identifies commonalities in helper functions would indeed be a good idea.  

I see that in `vdigest.R` you quite cleverly rewrote / compacted a few functions.  Can I maybe suggest a two step approach that might preserve more of the history?

First, we change `digest()` in `digest.R` to move the functions out.  The names you have in `vdigest.R` are generally sensible.   Once that is done, we maybe replace those functions in a second step (second PR?) with your rewrites.   Does that sound sane, or am I overcomplicating things?

Thanks again for the previous PR.  You are now also listed as a co-author among the ever growing list of contributors.

**Comment by mdequeljoe** [comment]
Thanks for the quick CRAN update. This approach sounds good to me.

**Comment by eddelbuettel** [comment]
[ CRAN update currently on hold as one package ([optmatch](http://cloud.r-project.org/package=optmatch)) came back with a failed test during reverse depends. I could not replicate that, and responded saying so.  We'll see what a human at CRAN says once they get to it.)

Thanks for the quick reply.  So between us we can probably just start a new branch changing `digest.R` first.  I may not get to it for a few days, but it isn't urgent.  Again thanks for the eagle-eyed approach at extending _and_ simplyfying.

**Comment by eddelbuettel** [comment]
Is there actually a second step we need to do or it this the main refactor?

One thing I was wondering was to maybe bring the `utils.c` file in and append to the end of `digest.c` just because that file has a copyright header etc pp (and I failed to keep track on a per-file basis of everybody who contributed....)

Thoughts?   Anything else you think we should before a next release?

**Comment by mdequeljoe** [comment]
Yes on second thought I think this may be sufficient as it is for now. And good point on the copyright header (for the `.R` files), so I think in this regard it would indeed make sense to append this back.

**Comment by eddelbuettel** [comment]
I can take care of it. Poke me here in a few days if I happen to forget about it (currently traveling and will have to catch up to a few other things too).  I'll then roll up a release and get your fine contribution to CRAN.

**Comment by eddelbuettel** [comment]
I added the copyright headers. I will need to double check I didn't forget people -- I based it in a first pass on  the commit logs.

But with that I think we can close the refactor ticket.  Thanks again for your `getVDigest()` PR!
